pallets/cash/src/reason.rs

```diff
@@ -60,6 +58,7 @@ pub enum Reason {
     SetYieldNextError(SetYieldNextError),
     SignatureAccountMismatch,
     SignatureMismatch,
+    TimestampMissing,
```

Comment:
I think we should only append to the end. Shuffling enums should be a no-go

Reply:
The `Reason` type isn't stored anywhere so it shouldn't break an upgrade (I think in general we should/could promise that about Reason now, although it did *used* to be in the Failed processing event state). I also think there's some benefit to grouping these and still dream of the day we might generate these codes. But I guess the argument is that it breaks (or could break) the client

Reply:
Though we've seen clients break~ish when they've used outdated reasons, which is harder that the reason comes up wrong/broken. But I could see this being a "safe to modify" enum if we wanted to go that way.